reference_entities/enrich_a_record_with_new_attribute_type.rst

```diff
@@ -0,0 +1,443 @@
+    use Akeneo\ReferenceEntity\Domain\Model\Attribute\AbstractAttribute;
+
+    class EditSimpleMetricValueCommandFactory implements EditValueCommandFactoryInterface
+    {
+        public function supports(AbstractAttribute $attribute, array $normalizedValue): bool
+        {
+             return
+                 $attribute instanceof SimpleMetricAttribute &&
+                '' !== $normalizedValue['data'] &&
+                is_string($normalizedValue['data']);
+        }
+
+        public function create(AbstractAttribute $attribute, array $normalizedValue): AbstractEditValueCommand
+        {
+            $command = new EditSimpleMetricValueCommand(
+                $attribute,
+                $normalizedValue['channel'],
+                $normalizedValue['locale'],
+                $normalizedValue['data']
+            );
+
+            return $command;
+        }
+    }
+
+Don't forget to register your factory to be recognized by our registry:
+
+.. code-block:: yaml
+
+    acme.application.factory.record.edit_simple_metric_value_command_factory:
+        class: Acme\CustomBundle\Application\Record\EditRecord\CommandFactory\EditSimpleMetricValueCommandFactory
+        tags:
+            - { name: akeneo_referenceentity.edit_record_value_command_factory }
+
+Now that we have our command, we need a specific value updater that will be able to understand this command to update a simple metric value:
+
+.. code-block:: php
+
+    <?php
+
+    namespace Acme\CustomBundle\Application\Record\EditRecord\ValueUpdater;
+
+    use Acme\CustomBundle\Application\Record\EditRecord\CommandFactory\EditSimpleMetricValueCommand;
+    use Acme\CustomBundle\Domain\Model\Record\Value\SimpleMetricData;
+    use Akeneo\ReferenceEntity\Application\Record\EditRecord\CommandFactory\AbstractEditValueCommand;
+    use Akeneo\ReferenceEntity\Application\Record\EditRecord\ValueUpdater\ValueUpdaterInterface;
+    use Akeneo\ReferenceEntity\Domain\Model\ChannelIdentifier;
+    use Akeneo\ReferenceEntity\Domain\Model\LocaleIdentifier;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Record;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\ChannelReference;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\LocaleReference;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\Value;
+
+    class SimpleMetricUpdater implements ValueUpdaterInterface
+    {
+        public function supports(AbstractEditValueCommand $command): bool
+        {
+            return $command instanceof EditSimpleMetricValueCommand;
+        }
+
+        public function __invoke(Record $record, AbstractEditValueCommand $command): void
+        {
+            if (!$this->supports($command)) {
+                throw new \RuntimeException('Impossible to update the value of the record with the given command.');
+            }
+
+            $attribute = $command->attribute->getIdentifier();
+            $channelReference = (null !== $command->channel) ?
```

Comment:
from an integrator point of view, I think it would make sens to have an helper here to avoid having to do the ternary